tools/mason/MasonBuild.chpl

```diff
@@ -0,0 +1,147 @@
+
+use TOML;
+use Spawn;
+use FileSystem;
+use mason;
+
+proc BuildProgram() {
+  if isFile("Mason.lock") {
+    if !isDir("target") {
+      // Make Target Directory
+      makeTargetFiles();
+    }
+
+    //Install dependencies into .mason/src
+    var toParse = open("Mason.lock", iomode.r);
+    var lockFile = parseToml(toParse);
+    var sourceList = genSourceList(lockFile);
+    getSrcCode(sourceList);
+
+    // Compile Program
+    if compileSrc(lockFile) {
+      writeln("Build Successful");
+    }
+    else {
+      writeln("Build Failed");
+    }
+
+    // Close memory
+     delete lockFile;
+     toParse.close();
+  }
+  else {
+    writeln("Cannot build: no Mason.lock found");
+  }
+} 
+
+
+/* Creates the rest of the project structure */
+proc makeTargetFiles() {
+  var makeTarget = "mkdir target";
```

Comment:
I don't think performance is crucial in this function, and using the library calls would be much cleaner.